### ISSUE TITLE
SessionStorage does not respond custom model names in 11.7.0

### ISSUE THREAD

**Issue by uurcankaya** [Shopify/shopify_app#862]
[ New session storage strategies](https://github.com/Shopify/shopify_app/blob/master/lib/shopify_app/session/storage_strategies/shop_storage_strategy.rb) do not take model names other than "Shop" into account. This seems like a breaking change.

Hence session storage does not respond 
`include ShopifyApp::SessionStorage`

The solution is to override the store and retrieve methods in the controller and remove the SessionStorage call. 
 
```
#include ShopifyApp::SessionStorage

#Override Shopify Session Storage
  def self.store(auth_session, *args)
   shop = Integration.find_or_initialize_by(shopify_domain: auth_session.domain)
   shop.shopify_token = auth_session.token
   shop.save!
   shop.id
 end

 def self.retrieve(id)
   return unless id
   if shop = Integration.find_by(id: id)
     ShopifyAPI::Session.new(
       domain: shop.shopify_domain,
       token: shop.shopify_token,
       api_version: shop.api_version
     )
   end
 end
```

A simple fix might be removing "Shop" [here](https://github.com/Shopify/shopify_app/blob/master/lib/shopify_app/session/storage_strategies/shop_storage_strategy.rb) or use "self"

```
shop = find_or_initialize_by(shopify_domain: auth_session.domain)
        shop.shopify_token = auth_session.token
        shop.save!
        shop.id
```

**Comment by tanema** [comment]
You are absolutely right, I will get that fixed right away

**Comment by tanema** [comment]
It seems like it is not quite as straightforward as doing that change as we no longer just extend the model when you include session storage. I am going to tag @ashchristopher and @slucaskim for this. 

I think we will need to re-arrange them a bit to be modules again and selectively include the module depending on the configuration.

**Comment by uurcankaya** [comment]
@tanema perhaps we can add it to the readme for the time being under "Upgrading to 11.7.0"

**Comment by tanema** [comment]
I am going to cut a release today with https://github.com/Shopify/shopify_app/pull/868 as fix

**Comment by tanema** [comment]
This has been shipped as a patch in 11.7.1

**Comment by uurcankaya** [comment]
👏
